cdap-data-fabric/src/main/java/co/cask/cdap/data2/dataset2/lib/table/hbase/HBaseTable.java

```diff
@@ -243,6 +276,19 @@ private Put getIncrementalPut(Put existing, byte[] row) {
       .build();
   }
 
+  private IncrementBuilder getIncrement(IncrementBuilder existing, byte[] row, boolean transactional)
+    throws IOException {
+    if (existing != null) {
```

Comment:
I don't think the builder is reusable as it never cleans the columns in the internal HBase `Increment` object. The same applies for the existing method that reuses the put builder.